ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java

```diff
@@ -31,6 +31,7 @@
   private final short accountId;
   private final short containerId;
   private final long operationTimeMs;
+  private final short updateVersion;
```

Comment:
The new update version scheme (which I'll add comments for) will update the version whenever a blob is undeleted, so if it's above zero, it implicitly shows that the blob was undeleted

Reply:
I feel `isUndeleted ` is needed, at least it's convenient. 

Because T1, D1 may be after U1,  which means U1, T1, D1 may in same version. updateVersion>0  doesn't mean U. 

Look at `isTtlUpdate` and `isDeleted` usage in replication. Replication also needs a way to recognize `isUndelete`. (Though `update>0 && !isTtlUpdate && !isDeleted` can be used, `isUndelete` makes coding easier.)

Reply:
Then you need to check both `isDelete` and `updateVersion` to distinguish subsequent delete  and undelete.

Reply:
When going through the code, I was under the impression that MessageInfo represented the state of a blob (represented by the original put plus any updates), not an object in-memory version of specific update blob (like a specific Delete blob).

Reply:
Do you guys think we need another name instead of `updateVersion`? Since 
1. we have `updateRecord`,  updateVersion may lead to confusion.  
2. Both Undelete and Update start with U...

Reply:
oh ok, I misread the intention here; because the comment highlighted the fields, I thought you were all suggesting adding another private field for storing whether the message had been undeleted.  I'm fine adding a method called `isUndeleted()` that checks the version and returns whether the node had been undeleted or not